source/reference/compatibility.txt

```diff
@@ -157,3 +157,14 @@ in ``mongosh`` better align with the types used by the MongoDB Drivers.
    For more information on managing types, refer to the
    :manual:`schema validation overview </core/schema-validation>`.
 
+``--eval`` Behavior
+-------------------
+
+The output of ``mongo --eval`` has a different format than the output
+of ``mongosh --eval``. 
```

Comment:
I feel like this is pretty vague and might leave users unnecessarily worrying about compatibility concerns.

Neither `mongo` nor `mongosh` provide output that is consistently programmatically parseable (e.g. if a BSON-specific object type is used) – it might be worth just being explicit here and saying that mongosh does not quote object keys in its output.